File: API/src/main/java/fontys/sem3/school/business/WatchlistService.java
package fontys.sem3.school.business;

import fontys.sem3.school.persistence.entity.WatchlistEntity;
import org.springframework.web.bind.annotation.PathVariable;

import java.util.List;

public interface WatchlistService {

    WatchlistEntity saveWatchlist(WatchlistEntity watchlist);
    List<WatchlistEntity> getWatchlistForUser(Long userid);
    List<WatchlistEntity> getWatchlists();
    WatchlistEntity updateWatchlist(Long watchlistId, WatchlistEntity watchlist);

    void deleteWatchlist(@PathVariable Long watchlistId);
}

File: front end/src/services/WatchlistService.js
import axios from 'axios';
import authHeader from './AuthenticationHeader';

const API_BASE_URL = 'http://localhost:8080';

const WatchlistService = {
  getWatchlists: () => {
    return axios.get(`${API_BASE_URL}/watchlist`);
  },
  getWatchlistForUser: () => {
    return axios.get(`${API_BASE_URL}/watchlist/personal`,{headers: authHeader()});
  },
  getWatchlistById: (id) => {
    return axios.get(`${API_BASE_URL}/watchlist/${id}`);
  },
  saveWatchlist: (watchlist) => {
    return axios.post(`${API_BASE_URL}/watchlist`, watchlist,{headers: authHeader()});
  },
  updateWatchlist: (watchlist) => {
    return axios.put(`${API_BASE_URL}/watchlist/${watchlist.id}`, watchlist);
  },
  deleteWatchlist: (watchlistId) => {
    return axios.delete(`${API_BASE_URL}/watchlist/${watchlistId}`);
  },
};

export default WatchlistService;
